torch/csrc/distributed/rpc/rpc_agent.h

```diff
@@ -157,7 +157,7 @@ class TORCH_API RpcAgent {
   // If ``message.isRequest()`` is true, the ``FutureMessage`` will be
   // completed when the response arrives. For other message types, the Future
   // should be ignored by the caller.
-  virtual std::shared_ptr<FutureMessage> send(
+  virtual std::shared_ptr<JitFuture> send(
```

Comment:
I've mostly seen JitFuture being passed around through an intrusive_ptr, rather than a shared_ptr. I guess that's somewhat more efficient (not sure how much). Should we consider it here too?

Reply:
@wanchaol @gmagogsfm is there any perf penalty if we use `ivalue::Future` with `std::shared_ptr`?

Reply:
There shouldn't be too much perf penalty, it's just that shared_ptr store the refcount in a separate memory space instead of embedded in the object, so from cache locality prospective intrusive_ptr should be a bit faster. We passed around JitFuture through intrusive_ptr also for the reason of making it to be able to held by `IValue` so that it's easier to bind to TorchScript if needed.

Reply:
I see, thanks for the explanation!